FSharpLu.Json/Compact.fs

```diff
@@ -18,43 +18,43 @@ module private ConverterHelpers =
         if System.Char.IsLower (name, 0) then name
         else string(System.Char.ToLower name.[0]) + name.Substring(1)
 
-    let inline memorise f = 
-        let d = Dictionary<_, _>()
-        fun key -> 
-            match d.TryGetValue(key) with
-            | (true, v) -> v
-            | (false, _) -> 
+    let inline memorise f =
+        let cache = Dictionary<_, _>()
+        fun key ->
+            match cache.TryGetValue(key) with
+            | true, v -> v
+            | false, _ ->
                 let result = f key
-                d.[key] <- result
+                cache.[key] <- result
                 result
 
-    let getUnionCaseFieldsMemorised = memorise FSharpValue.PreComputeUnionReader
-    let getUnionTagMemorised = memorise FSharpValue.PreComputeUnionTagReader
-    let getUnionCasesByTagMemorised = memorise (fun t -> FSharpType.GetUnionCases(t) |> Array.map (fun x -> x.Tag, x) |> dict)
-    
-    let getUnionCasesMemorised = memorise (fun (t: Type) -> FSharpType.GetUnionCases t)
+    let inline private getUnionCaseFields t = memorise FSharpValue.PreComputeUnionReader t
+    let inline getUnionTag t = memorise FSharpValue.PreComputeUnionTagReader t
+    let inline private getUnionCasesByTag t = memorise (fun t -> FSharpType.GetUnionCases(t) |> Array.map (fun x -> x.Tag, x) |> dict) t
+    let inline getUnionCases t = memorise (fun (t: Type) -> FSharpType.GetUnionCases t) t
 
-    let getUnionTagOfValueMemorised v = 
+    let inline private getUnionTagOfValue v =
         let t = v.GetType()
-        getUnionTagMemorised t v
+        getUnionTag t v
 
     let inline getUnionFieldsMemorised v =
-        let cases = getUnionCasesByTagMemorised (v.GetType())
-        let tag = getUnionTagOfValueMemorised v
+        let cases = getUnionCasesByTag (v.GetType())
```

Comment:
Watching your repo - curious why are you changing them from values into inline functions? For the case I was using this for I would think this would be a big performance regression since the memorisation dictionary will be created each invocation effectively rendering the memorisation useless? Unless I'm not reading the code right.
I left the functions that called "memorise" as values so the dictionary would cache the results for subsquent invocations.